util/unifyfs/src/unifyfs-rm.c

```diff
@@ -922,30 +988,79 @@ int unifyfs_start_servers(unifyfs_resource_t* resource,
         fprintf(stderr, "failed to create server launch server process (%s)\n",
                         strerror(errno));
         return -errno;
-    } else if (pid == 0) {
+    }
+
+    args->arg_type = UNIFYFS_CONSTRUCT_ARGS_UNIFYFSD;
+    if (pid == 0) {
         if (args->script != NULL) {
-            return script_launch(resource, args);
+            unifyfsd_return_val = script_launch(resource, args);
         } else {
-            return resource_managers[resource->rm].launch(resource, args);
+            unifyfsd_return_val =
+	      resource_managers[resource->rm].launch(resource, args);
+        }
+    } else {
+        unifyfsd_return_val = wait_server_initialization(resource, args);
+        if (unifyfsd_return_val) {
+	    fprintf(stderr,
+		    "ERROR: failed to wait for server initialization: (%d s)\n",
+		    unifyfsd_return_val);
         }
     }
 
-    rc = wait_server_initialization(resource, args);
-    if (rc) {
-        fprintf(stderr, "ERROR: failed to wait for server initialization\n");
+    if (args->stage_in) {
+        if (args->script) {
+	    fprintf(stderr,
+		    "Simultaneous stage_in and script options not supported yet.\n");
+            return -EINVAL;
+        }
+        if (args->transfer_exe == NULL) {
+            // this should never ever happen, but to avoid a seg fault
+            fprintf(stderr,
+		    "ERROR!  stage_in set but transfer_exe is NULL!!!\n");
+            return -EINVAL;
+        }
+        args->arg_type = UNIFYFS_CONSTRUCT_ARGS_STAGE_IN;
+        transfer_return_val =
+	  resource_managers[resource->rm].launch(resource, args);
+	// If there's any possibility that the stage-in happens asynchronously,
+	// here's where we would put the check to wait until it was all done.
     }
 
-    return rc;
+    // Possibly we should combine the two return values in the future?
+    transfer_return_val++;
+    transfer_return_val--;
```

Comment:
Can these be removed?

Reply:
Yes.  Sorry.  I put them in to get the compiler to stop whining about an unused variable.  I meant to fix that before I uploaded.  I'll fix that in the next commit.